compiler/dsl/src/json_export.rs

```diff
@@ -0,0 +1,2093 @@
+    let mut start_line = 1u32;
+    let mut start_column = 1u32;
+    let mut end_line = 1u32;
+    let mut end_column = 1u32;
+    
+    let mut found_start = false;
+    let mut found_end = false;
+    
+    for (byte_index, ch) in source_text.char_indices() {
+        // Check if we've reached the start position
+        if !found_start && byte_index >= start_byte {
+            start_line = line;
+            start_column = column;
+            found_start = true;
+        }
+        
+        // Check if we've reached the end position
+        if !found_end && byte_index >= end_byte {
+            end_line = line;
+            end_column = column;
+            found_end = true;
+            break;
+        }
+        
+        // Update line and column based on character
+        if ch == '\n' {
+            line += 1;
+            column = 1;
+        } else {
+            column += 1;
+        }
+    }
+    
+    // If we didn't find the end position, use the final line/column
+    if !found_end {
+        end_line = line;
+        end_column = column;
+    }
+    
+    (start_line, start_column, end_line, end_column)
+}
+
+/// Serializable representation of a symbol table entry.
+#[derive(Debug, Clone, Serialize)]
+pub struct SymbolEntry {
+    pub name: String,
+    pub symbol_type: String,
+    pub scope: String,
+    pub declaration_location: Option<SerializableSourceSpan>,
+    pub data_type: Option<String>,
+    pub is_external: bool,
+}
+
+/// Serializable representation of the symbol table.
+#[derive(Debug, Clone, Serialize)]
+pub struct SymbolTable {
+    pub symbols: std::collections::HashMap<String, SymbolEntry>,
+    pub scopes: Vec<String>,
+}
+
+/// Serializable representation of a comment.
+#[derive(Debug, Clone, Serialize)]
+pub struct CommentInfo {
+    pub text: String,
+    pub style: String, // "block" for (* *) or "line" for //
+    pub location: Option<SerializableSourceSpan>,
+    pub attached_to: Option<String>, // ID of the AST node this comment is attached to
+}
+
+/// Serializable representation of formatting information.
+#[derive(Debug, Clone, Serialize)]
+pub struct FormattingInfo {
+    pub comments: Vec<CommentInfo>,
+    pub whitespace_significant: bool,
+}
+
+impl SymbolTable {
+    pub fn new() -> Self {
+        Self {
+            symbols: std::collections::HashMap::new(),
+            scopes: vec!["global".to_string()],
+        }
+    }
+
+    pub fn add_symbol(&mut self, id: String, entry: SymbolEntry) {
+        self.symbols.insert(id, entry);
+    }
+
+    pub fn add_scope(&mut self, scope: String) {
+        if !self.scopes.contains(&scope) {
+            self.scopes.push(scope);
+        }
+    }
+}
+
+/// Extract unique source file IDs from a Library AST.
+/// This function provides a simple implementation that returns a default file list.
+/// In a full implementation, this would traverse the AST to collect all unique file IDs.
+fn extract_source_files(_library: &Library) -> Vec<String> {
```

Comment:
Question - why not operate on a `project` rather than on `library`. A project contains multiple files and knows what files are relevant. The library is an intermediate representation and is modified through multiple stages?

To have this work on a project you would need to move this into plc2x because that's where project is defined.